R-package/tests/testthat/test_lgb.Booster.R

```diff
@@ -1653,6 +1654,24 @@ test_that("Booster's print, show, and summary work correctly", {
     .check_methods_work(model)
 })
 
+test_that("Booster's print, show, and summary work correctly when objective is not provided", {
+  data("iris")
+  model <- lgb.train(
+      data = lgb.Dataset(
+          as.matrix(iris[, seq_len(3L)])
+          , label = iris[, 4L]
+      )
+      , verbose = .LGB_VERBOSITY
+      , nrounds = 5L
+      , params = list(num_threads = .LGB_MAX_THREADS)
+  )
+
+  log_txt <- capture.output(print(model))
+  expect_true(any("Objective: (default)" == log_txt))
```

Comment:
```suggestion
  expect_true(any(log_txt == "Objective: (default)"))
```

This is a very minor, personal-preference thing, but comparing a character vector to a string with `==`, I find it a bit easier to understand spelled like this. In my head, this is "any of the elements of the vector are equal to this string" instead of "this string is equal to any elements in the vector". That's why I suggested that type of spelling in https://github.com/microsoft/LightGBM/pull/6850/files#r1982618196

I'm going to apply this change, then merge this PR once CI runs one more time.